source/change-streams.rst

```diff
@@ -339,19 +422,46 @@ Single Server Topologies
 
 Presently, change streams cannot be initiated on single server topologies as they do not have an oplog.  Drivers MUST NOT throw an exception in this scenario, but instead rely on an error returned from the server.  This allows for the server to seamlessly introduce support for this in the future, without need to make changes in driver code.
 
+startAtClusterTime
+^^^^^^^^^^^^^^^^^^
+
+:since: 4.0
+
+``startAtClusterTime`` specifies that a ``changeStream`` will only return changes that occurred after the specified timestamp.
+
+``startAtClusterTime`` and ``resumeAfter`` are mutually exclusive; if both ``startAtClusterTime`` and ``resumeAfter`` are set, the server will return an error. Drivers MUST NOT throw a custom error, and MUST defer to the server error.
+
+If neither ``startAtClusterTime`` or ``resumeAfter`` are specified, and the server version is >= ``4.0`` drivers MUST set a default ``startAtClusterTime`` value using a ``$clusterTime`` from any server response (ex: ``ismaster``). This allows change streams to be resumed before the first notification is received.
+
 resumeAfter
 ^^^^^^^^^^^
 
-When resuming a change stream after a disconnect, the driver issuing a new ``$changeStream`` request MUST specify a ``resumeAfter`` key with a resume token from the last change it saw.  In this case, the aggregation will return notifications starting with the log entry immediately *after* the provided token.  If the resume token specified does not exist, the server will return an error.  If ``resumeAfter`` is omitted completely, or is null, the most recent oplog entry will be returned.
+:since: 3.6
+
+``resumeAfter`` is used to resume a changeStream that has been stopped to ensure that only changes starting with the log entry immediately *after* the provided token will be returned. If the resume token specified does not exist, the server will return an error. 
 
 Resume Process
 ^^^^^^^^^^^^^^
 
-Once a ``ChangeStream`` has encountered a resumable error, it MUST attempt to resume one time.  The process for resuming MUST follow these steps:
+Once a ``ChangeStream`` has encountered a resumable error, it MUST attempt to resume one time. The process for resuming MUST follow these steps:
+
+- Perform server selection.
+- Connect to selected server.
+- If the ``ChangeStream`` has not received any changes, and ``resumeAfter`` is not specified, and the server version is >= ``4.0``:
+
+    - The driver MUST execute the known aggregation command.
+    - The driver MUST specify the ``startAtClusterTime`` key set to the original cluster time.
+    - The driver MUST NOT set a ``resumeAfter`` key.
+    - In this case, the ``ChangeStream`` will return all changes that occurred after the specified ``startAtClusterTime``.
+- If the ``ChangeStream`` has received any changes, or the server version is < ``4.0``, or a ``resumeAfter`` token is specified by the user:
```

Comment:
Since the second condition covers all seven of the other rows of the truth table for the three variables, it's clearer to just say "Otherwise" or "Else" here instead of the more complicated boolean expression that's here now.

Reply:
done